synapse/tests/test_cortex.py

```diff
@@ -154,6 +154,53 @@ def test_onadd(self):
                 node = snap.addNode('inet:ipv4', '1.2.3.4')
                 self.eq(node.buid, func.args[0].buid)
 
+    def test_cancel_cortex(self):
+        with self.getTestCore() as core:
+            self._run_cancel_assertions(core)
+
+    def test_cancel_coreapi(self):
+        with self.getTestDmon(mirror='dmoncore') as dmon:
+            core = dmon._getTestProxy('core')
+            self._run_cancel_assertions(core)
+
+    def _run_cancel_assertions(self, core):
+        # Add some nodes
+        N = 1000
+        nodes = [(('inet:ipv4', i), {}) for i in range(N)]
+        list(core.addNodes(nodes))
+
+        # Start executing the query but don't iterate yet
+        init = None
+        genr = core.storm('inet:ipv4')
+        if isinstance(genr, s_telepath.Genr):
+            for msg in genr:
+                init = msg
+                break
+        else:
+            init = next(genr)
+        guid = init[1].get('guid')
+
+        # Cancel the query before iterating further
+        self.true(core.cancelQueryByGuid(guid))
+
+        # Start iterating, make sure Canceled is raised
```

Comment:
I get different results here against cortex vs coreapi. Should the behave differently?

Against Cortex, I get the canceled error message and a fini message. Against CoreApi, I just get no more messages.